src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -230,7 +233,20 @@ public async Task SslStream_UntrustedCaWithCustomCallback_OK()
                 };
 
             var serverOptions = new SslServerAuthenticationOptions();
-            serverOptions.ServerCertificateContext = SslStreamCertificateContext.Create(_serverCert, _serverChain);
+
+            X509Certificate2Collection serverChain;
+            if (usePartialCHain)
+            {
+                // give certificates  without root
+                serverChain = new X509Certificate2Collection();
+                serverChain.Add(_serverChain[0]);
```

Comment:
Is it worth adding another case here for more than 1 cert but still a partial chain?

Reply:
I'm not sure at this point, perhaps @bartonjs may have some inout. All the ssl  tests now use either self-signed cert or chain with one intermediate and root.

Reply:
Chains can be arbitrarily long, I'd consider doing one that's 5 (root, intermed1, intermed2, intermed3, leaf) to make sure you're not special casing that "partial" means zero or one items (in the 5 case it could be any of 0-3)

Reply:
ok. I'll add that to the test helper.

Reply:
I added chain as @bartonjs suggested. I chop it at random point to validate any fixed assumptions.